app/src/lib/dispatcher/error-handlers.ts

```diff
@@ -287,9 +287,11 @@ export async function mergeConflictHandler(
     switch (command) {
       case 'pull':
         dispatcher.recordMergeConflictFromPull()
+        dispatcher.mergeConflictDetected()
```

Comment:
Is there a reason why we need to make these calls alongside existing conflict-related calls? Could `Dispatcher.recordMergeConflictFromPull` and `Dispatcher. recordMergeConflictFromExplicitMerge` also invoke `this.appStore._mergeConflictDetected()` internally?

Reply:
I can't really speak on calling `dispatcher. recordMergeConflictFromExplicitMerge()`, but I can explain the reasoning for putting the call `dispatcher.mergeConflictDetected()` in both cases: the method names `recordMergeConflictFromPull` and `recordMergeConflictFromExplicitMerge` doesn't hint that it should do anything else other than increment our stats counters.

Reply:
> doesn't hint that it should do anything else other than increment our stats counters

Should they? I think that's the part I was misunderstanding, so thanks for clarifying.

Thinking more on this, this feels like a chance to encapsulate related behaviour and make the dispatcher reflect how we want to use it, rather than following the old pattern of it being 1-1 with `AppStore`.

We could end up with a more readable flow, like this:

```ts
switch (command) {
  case 'pull':
    dispatcher.mergeConflictDetectedFromPull()
    break
  case 'merge':
    dispatcher.mergeConflictDetectedFromExplicitMerge()
    break
}
```

The dispatcher functions could then look like this:

```
/**
 * Updates the application state to indicate a conflict is in-progress,
 * and increments the relevant metric.
 */
public mergeConflictDetectedFromExplicitMerge() {
  this.appStore._mergeConflictDetected()
  return this.statsStore.recordMergeConflictFromExplicitMerge()
}
```

Thoughts?

Reply:
I like it. It's definitely more descriptive of what we're doing here. #ReduxStateOfMind! 😉